src/OrchardCore/Orchard.Environment.Shell/Builders/ShellContainerFactory.cs

```diff
@@ -87,16 +87,18 @@ public IServiceProvider CreateContainer(ShellSettings settings, ShellBlueprint b
                 var feature = blueprint.Dependencies.FirstOrDefault(x => x.Key == startup.GetType()).Value.FeatureInfo;
 
                 ServiceCollection featureServiceCollection;
-                ServiceCollection startupServices = new ServiceCollection();
-
+                
                 if (!featureServiceCollections.TryGetValue(feature, out featureServiceCollection))
                 {
                     featureServiceCollections.Add(feature, featureServiceCollection = new ServiceCollection());
                 }
 
-                startup.ConfigureServices(startupServices);
-                tenantServiceCollection.Add(startupServices);
-                featureServiceCollection.Add(startupServices);
+                int previousTenantServiceCount = tenantServiceCollection.Count;
+                startup.ConfigureServices(tenantServiceCollection);
```

Comment:
Please note that I am now passing tenantServiceCollection to the ConfigureServices method instead of startupServices. This is the main change/fix.